### DISCUSSION TITLE
How to use `maxEdgeLen` with `generateTileCache`?

### DISCUSSION THREAD

**Discussion by rob-myers** [isaac-mason/recast-navigation-js#308] [General]
The example function for TileCache [`generateTileCache`](https://github.com/isaac-mason/recast-navigation-js/blob/d3adca721ce1993976d4cc01881a61e76e9ecec3/packages/recast-navigation-generators/src/generators/generate-tile-cache.ts#L124) does not make use of `config.maxEdgeLen`.

How can we add this constraint?

**Comment by isaac-mason** [comment]
Hey @rob-myers,

It seems TileCache doesn't use the maxEdgeLen config, there's no reference to maxEdgeLen being used in the `Sample_TempObstacles` example from the recastnavigation repo: https://github.com/recastnavigation/recastnavigation/blob/599fd0f023181c0a484df2a18cf1d75a3553852e/RecastDemo/Source/Sample_TempObstacles.cpp#L1235

In the other generators maxEdgeLen is used with `rcBuildContours` / `buildContours`, but [DetourTileCache has seperate logic for building contours](https://github.com/isaac-mason/recastnavigation/blob/599fd0f023181c0a484df2a18cf1d75a3553852e/DetourTileCache/Source/DetourTileCacheBuilder.cpp#L748) that doesn't appear to take a maxEdgeLen.

**Comment by rob-myers** [comment]
Oh well, thanks for the explanation.

**Reply by isaac-mason** [reply]
No worries! I think we could fix the typescript types so it doesn't seem like you can use the parameter.
